src/main/java/com/example/amazon/reviews/service/impl/ReviewServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.example.amazon.reviews.service.impl;
+
+import com.example.amazon.reviews.mapper.ReviewMapper;
+import com.example.amazon.reviews.model.Product;
+import com.example.amazon.reviews.model.Review;
+import com.example.amazon.reviews.model.dto.ReviewDto;
+import com.example.amazon.reviews.repository.ReviewRepository;
+import com.example.amazon.reviews.service.ReviewService;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ReviewServiceImpl implements ReviewService {
+    private static final String WORD_DELIMITER_REGEXP = "[^a-zA-Z]+";
```

Comment:
for example the sentence  "it's a good product" would be divided to  words: "it" , "s", "a", "good", "product"?